src/App.tsx

```diff
@@ -17,18 +30,35 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filter={filter}
+                setFilter={setFilter}
+                searchQuery={search}
+                setSearch={setSearch}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!todos.length && <Loader />}
+
+              <TodoList
+                todos={todos}
+                selectedTodo={selectedTodoId}
+                selectTodo={setTodoId}
+                filter={filter}
+                search={search}
+              />
             </div>
           </div>
         </div>
       </div>
-
-      <TodoModal />
+      {selectedTodoId && (
+        <TodoModal
+          todos={todos}
+          selectedTodo={selectedTodoId}
+          unselectTodo={() => setTodoId(0)}
```

Comment:
as I see you use null value by deafault 
```suggestion
          unselectTodo={() => setTodoId(null)}
```